sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -643,6 +647,34 @@ object DataSourceStrategy {
     (nonconvertiblePredicates ++ unhandledPredicates, pushedFilters, handledFilters)
   }
 
+  def translateAggregate(aggregates: AggregateExpression): Option[AggregateFunc] = {
+
+    def columnAsString(e: Expression): String = e match {
+      case AttributeReference(name, _, _, _) => name
+      case Cast(child, _, _) => child match {
+        case AttributeReference(name, _, _, _) => name
+        case _ => ""
+      }
+      case _ => ""
+    }
+
+    aggregates.aggregateFunction match {
```

Comment:
What if `aggregates` has `isDistinct=true` or `filter`?

Reply:
I will need to change the following to add `isDistinct` and `filter`. Also change `translateAggregate` accordingly. When push down the aggregates, need to check the filter to make sure it can be pushed down too.
```
case class Avg(column: String, isDistinct: Boolean, filter: Option[Filter]) extends AggregateFunc

case class Min(column: String, isDistinct: Boolean, filter: Option[Filter]) extends AggregateFunc

case class Max(column: String, isDistinct: Boolean, filter: Option[Filter]) extends AggregateFunc

case class Sum(column: String, isDistinct: Boolean, filter: Option[Filter]) extends AggregateFunc
```